Test/test2/IncompleteArithmetic-git-issue-321.bpl.expect

```diff
@@ -0,0 +1,5 @@
+/Users/leino/Source/boogie/Test/test2/IncompleteArithmetic-git-issue-321.bpl(11,3): Error BP5001: This assertion might not hold.
+Execution trace:
+    /Users/leino/Source/boogie/Test/test2/IncompleteArithmetic-git-issue-321.bpl(11,3): anon0
+
+Boogie program verifier finished with 0 verified, 1 error
```

Comment:
The PR description says:

```
With my edit to Source/Provers/SMTLib/SMTLibProcess.cs, Boogie reports 0 errors (and also 0 verified) on the repro example in #321:

$ boogie -proverOpt:O:smt.arith.solver=2 IncompleteArithmetic-git-issue-321.bpl

Boogie program verifier finished with 0 verified, 0 errors
```

which contradicts what I am seeing in the .expect file.  What am I not understanding?

Reply:
I think the `.expect` file shows (something close to) what we want to get as an output. But the changes in the PR do not yet produce this desired result.

Reply:
Correct. Sorry for the confusion. The `.expect` file I gave is what I (think I) expect the result to be, but my PR does not achieve that result.

Reply:
@RustanLeino : Recall that Boogie's counterexample generation is now based on querying the model (rather than labels as was the case earlier).  Since in your example a model is not produced, it is unreasonable to expect a counterexample in the .expect file.

Reply:
Actually, I see that .expect file has changed again to "1 verified, 0 errors", which also does not make any sense to me since I am expecting that the version of Z3 supported by Boogie (presumably 4.8.9) is unable to verify this example.

Reply:
The error happens with Z3 v.4.8.9. I was about to check in the corresponding .expect file when I noticed that the Boogie build uses Z3 v.4.8.8. That version of Z3 _is_ is able to the assertion, and hence I changed the .expect file to reflect the "1 verified, 0 errors".

I have also tested the behavior with Z3 v.4.8.9, which returned "unknown" and has no model to show for. This PR includes code that attempts to report the error. I may not have done that in the best way. It seems that `CalculatePath` silently swallows any unexpected communication with Z3, which can result in it returning an empty path.